packages/relay/src/lib/decorators/rpcMethod.decorator.ts

```diff
@@ -20,12 +20,17 @@ export const RPC_METHOD_KEY = 'hedera-rpc-method';
  * }
  * ```
  *
- * @param _target - The prototype of the class (ignored in this implementation)
- * @param _propertyKey - The name of the method being decorated (ignored in this implementation)
- * @param descriptor - The property descriptor for the method
- * @returns The same property descriptor, allowing for decorator composition
+ * @param target - The method function
+ * @param context - The decorator context
+ * @returns The method function with RPC metadata attached
  */
-export function rpcMethod(_target: any, _propertyKey: string, descriptor: PropertyDescriptor) {
-  descriptor.value[RPC_METHOD_KEY] = true;
-  return descriptor;
+export function rpcMethod(target: any, context: ClassMethodDecoratorContext): any {
+  context.addInitializer(function (this: any) {
+    const methodName = String(context.name);
+    if (this[methodName]) {
+      this[methodName][RPC_METHOD_KEY] = true;
+    }
+  });
```

Comment:
is it possible to write it like this instead?

```suggestion
  target[RPC_METHOD_KEY] = true;  
```

Reply:
I didn't do it originally because the cache decorator used it, which replaced the original method and broke the other decorators. However your comment is correct and I can also remove it from there safely. It's removed from all decorators and all tests pass now